org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/internal/gradle/checksums/WrapperValidator.java

```
@@ -0,0 +1,97 @@
+
+
+	public WrapperValidator() {
+		this(new HashProvider());
+	}
+
+	public WrapperValidator(HashProvider hashProvider) {
+		this.hashProvider = hashProvider;
+	}
+
+	public String checkWrapper(String baseDir) throws IOException, NoSuchAlgorithmException {
+		if (DISABLED) {
+			return null;
+		}
+		Path wrapperJar = Paths.get(baseDir, GRADLE_WRAPPER_JAR);
+		if (!wrapperJar.toFile().exists()) {
+			return wrapperJar.toString() + " doesn't exist.";
+		}
+		Path wrapperProperties = Paths.get(baseDir, GRADLE_WRAPPER_PROPERTIES);
+		if (!wrapperProperties.toFile().exists()) {
+			return wrapperProperties.toString() + " doesn't exist.";
+		}
+		try (InputStream is = new FileInputStream(wrapperProperties.toFile())) {
+			Properties props = new Properties();
+			props.load(is);
+			String distributionUrl = props.getProperty(DISTRIBUTION_URL);
+			if (distributionUrl == null) {
+				return "Invalid the distributionUrl property";
+			}
+			String sha256Url = distributionUrl.replace("-bin.zip","").replace("-all.zip", "") + "-wrapper.jar.sha256";
+			String checksum = hashProvider.getChecksum(wrapperJar.toFile());
+			String remoteChecksum = downloadRequestsCache.getIfPresent(sha256Url);
+			if (remoteChecksum == null) {
+				URL url = new URL(sha256Url);
+				BufferedReader in = null;
+				try {
+					URLConnection urlConnection = url.openConnection();
+					in = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
+					remoteChecksum = in.readLine();
+					downloadRequestsCache.put(sha256Url, remoteChecksum);
+				} catch (IOException e) {
+					JavaLanguageServerPlugin.logException("sha256url='" + sha256Url + "'. You can try to add '-Dgradle.wrapper.check.disabled=true' to 'java.jdt.ls.vmargs'", e);
+				} finally {
+					if (in != null) {
+						in.close();
+					}
+				}
+			}
+			if (remoteChecksum != null && remoteChecksum.equals(checksum)) {
+				return null;
+			} else {
```

I'd rather throw an exception here

---

Fixed.